python/TestHarness/unit_tests/test_HarnessTester.py

```diff
@@ -95,3 +95,21 @@ def testDuplicateOutputsOK(self):
         self.assertNotRegexpMatches(output, 'heavy_out.e')
         # all
         self.assertNotRegexpMatches(output, 'FATAL TEST HARNESS ERROR')
+
+    def testDryRun(self):
+        """
+        Test that --dry-run returns a successful 'dry run' test
+        """
+        output = self.runTests('-i', 'diffs', '--dry-run')
+
+        self.assertRegexpMatches(output, 'test_harness\.exodiff.*?DRY RUN')
```

Comment:
Maybe also check the `csvdiff` one as well?
Also, to check that they are actually passing, the output should probably contain a "`^2 passed`"